It passed BE UT but failed on linking `libstarcache.a`.

`libstarcache.a` from `docker.io/starrocks/starlet-artifacts-centos7:v3.1-rc5` was compiled using fmt-7.
@kevincai It looks like you are the expert on starlet. How to compile starlet-artifacts with the fmt-8.1.1 and what would be the next step for this PR?

Thanks,
Jinho

> /var/local/thirdparty/installed/starlet/starlet_install/lib/libstarcache.a(disk_space_manager.cpp.o):
> In function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > fmt::v7::format<char [9], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long, char>(char const (&) [9], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long&&)':
/var/local/thirdparty/installed/starlet/third_party/include/fmt/core.h:1836: undefined reference to `fmt::v7::detail::vformat[abi:cxx11](fmt::v7::basic_string_view<char>, fmt::v7::format_args)'

--- a/docker/dockerfiles/dev-env/dev-env.Dockerfile
+++ b/docker/dockerfiles/dev-env/dev-env.Dockerfile
@@ -22,7 +22,7 @@ ARG predownload_thirdparty=false
 ARG thirdparty_url=https://cdn-thirdparty.starrocks.com/starrocks-thirdparty-main-20230317.tar
 ARG commit_id
 # check thirdparty/starlet-artifacts-version.sh, to get the right tag
-ARG starlet_tag=v3.1-rc5
+ARG starlet_tag=v3.1-rc6
 # build for which linux distro: centos7|ubuntu
 ARG distro=ubuntu
 

--- a/thirdparty/starlet-artifacts-version.sh
+++ b/thirdparty/starlet-artifacts-version.sh
@@ -9,4 +9,4 @@
 #   https://hub.docker.com/r/starrocks/starlet-artifacts-centos7/tags
 #
 # Update the following tag when STARLET releases a new version.
-export STARLET_ARTIFACTS_TAG=v3.1-rc5
+export STARLET_ARTIFACTS_TAG=v3.1-rc6

